src/jsonapi-datastore.js

```diff
@@ -11,10 +11,68 @@
     this.id = id;
     this._type = type;
     this._attributes = [];
+    this._dependents = [];
     this._relationships = [];
   }
 
   /**
+   * Add a dependent to a model.
+   * @method _addDependence
+   * @param {string} type The type of the dependent model.
+   * @param {string} id The id of the dependent model.
+   * @param {string} key The name of the relation found on the dependent model.
+    */
+  _addDependence(type, id, key) {
+    var self = this,
+      found;
+
+    found = self._dependents.find(function(dependent) {
+      return dependent.id === id && dependent.type === type && dependent.relation === key;
+    });
+    if (found === undefined) {
+      self._dependents.push({id: id, type: type, relation: key});
+    }
+  }
+
+  /**
+   * Removes a dependent from a model.
+   * @method _removeDependence
+   * @param {string} type The type of the dependent model.
+   * @param {string} id The id of the dependent model.
+    */
+  _removeDependence(type, id) {
+    var self = this,
+      found;
+
+    self._dependents.forEach(function(val, idx) {
+      if (val.id === id && val.type === type) {
+        self._dependents.splice(idx, 1);
+      }
+    });
+  }
+
+  /**
+   * Removes a relationship from a model.
+   * @method removeRelationship
+   * @param {string} type The type of the dependent model.
+   * @param {string} id The id of the dependent model.
+   * @param {string} relName The name of the relationship.
+    */
+  removeRelationship(type, id, relName) {
+    var self = this;
+    self._removeDependence(type, id);
```

Comment:
By doing this we are effectively removing all dependences to the model, although we should probably just remove the dependence for the current relationship (which implies that `_removeDependence` would have to take a `key` parameter as well).